compiler/modules/front/expr.bal

```diff
@@ -1224,8 +1224,24 @@ function codeGenFunctionCallExpr(ExprContext cx, bir:BasicBlock bb, s:FunctionCa
     }
     bir:BasicBlock curBlock = bb;
     bir:Operand[] args = [];
+    s:Expr[] restArgs = [];
+    t:SemType? restParamType = func.signature.restParamType;
     foreach int i in 0 ..< expr.args.length() {
-        var { result: arg, block: nextBlock } = check codeGenArgument(cx, curBlock, expr, func, i);
+        if restParamType != () && i >= func.signature.paramTypes.length() - 1 {
+            restArgs.push(expr.args[i]);
+        }
+        else {
+            var { result: arg, block: nextBlock } = check codeGenArgument(cx, curBlock, expr, func, i);
+            curBlock = nextBlock;
+            args.push(arg);
+        }
+    }
+    if restParamType != () {
+        Position startPos = restArgs.length() > 0 ? restArgs[0].startPos : expr.openParenPos;
```

Comment:
Turn these two lines into an if/else where you test restArgs.length() once.

Reply:
fixed